client/state/data-layer/wpcom/reader/teams/index.js

```diff
@@ -4,20 +4,26 @@
 import { READER_TEAMS_REQUEST, READER_TEAMS_RECEIVE } from 'state/action-types';
 import wpcom from 'lib/wp';
 
-export function handleTeamsRequest( state, action, next ) {
-	wpcom.req.get( '/read/teams', { apiVersion: '1.2' } )
+export function handleTeamsRequest( store, action, next ) {
+	return wpcom.req.get( '/read/teams', { apiVersion: '1.2' } )
 		.then(
-			payload => state.dispatch( {
-				type: READER_TEAMS_RECEIVE,
-				payload,
-			} ),
-			error => state.dispatch( {
-				type: READER_TEAMS_RECEIVE,
-				payload: error,
-				error: true,
-			} )
+			payload => {
+				store.dispatch( {
+					type: READER_TEAMS_RECEIVE,
+					payload,
+				} );
+				next( action );
+
+			},
+			error => {
+				store.dispatch( {
+					type: READER_TEAMS_RECEIVE,
+					payload: error,
+					error: true,
+				} );
+				next( action );
+			}
 		);
-	next( action );
```

Comment:
Moving `next( action )` into the promise continuations is probably a step backward here. We almost certainly want our actions to hit the reducers in a synchronous fashion. In this case, the reducers won't even be aware of the actions until the HTTP request succeeds or fails, which could be seconds of delay…